spec/brainstem-model-spec.coffee

```diff
@@ -94,14 +139,48 @@ describe 'Brainstem.Model', ->
       expect(model.fetch()).toEqual(promise)
 
     describe 'integration', ->
-      it 'something', ->
-        task = buildTask()
-        respondWith(server, '/api/tasks/1', resultsFrom: 'tasks', data: task)
+      it 'fetches model', ->
+        updatedModel = buildTask()
+
+        respondWith(server, "/api/tasks/#{model.id}", resultsFrom: 'tasks', data: updatedModel)
 
         model.fetch()
         server.respond()
 
-        expect(model.attributes).toEqual(task.attributes)
+        expect(model.attributes).toEqual(updatedModel.attributes)
+
+      it 'caches new model reference on fetch', ->
+        newTask = buildTask()
+
+        respondWith(server, "/api/tasks/#{newTask.id}", resultsFrom: 'tasks', data: newTask)
+
+        newModel = new App.Models.Task(id: newTask.id)
+        newModel.fetch()
+
+        server.respond()
+
+        expect(base.data.storage('tasks').get(newModel.id)).toEqual(newModel)
```

Comment:
Can this assert `cid` equality instead? It would be a bit stricter on the reference assertion.